docs/develop/best-practices.md

```diff
@@ -0,0 +1,164 @@
+---
+title: 'Best practices in Office Scripts'
+description: 'How to prevent common problems and write robust Office Scripts that can handle unexpected input or data.'
+ms.date: 05/10/2021
+localization_priority: Normal
+---
+
+# Best practices in Office Scripts
+
+There are some patterns you can follow in your Office Scripts to help ensure they run successfully. These should help you avoid common pitfalls as you start automating your Excel workflow.
+
+## Check if the object exists before using it
+
+Scripts often rely on a certain worksheet or table being present in the workbook. However, they might get renamed or removed between script runs. By checking if those tables or worksheets exist before calling methods on them, you can make sure the script doesn't end abruptly.
+
+The following sample code checks if the "Index" worksheet is present in the workbook. If the worksheet is present, the script gets a range and proceeds. If it isn't present, the script logs a custom error message.
+
+```TypeScript
+// Make sure the "Index" worksheet exists before using it.
+let indexSheet = workbook.getWorksheet('Index');
+if (indexSheet) {
+  let range = indexSheet.getRange("A1");
+  // Continue using the range...
+} else {
+  console.log("Index sheet not found.");
+}
+```
+
+You can also use the TypeScript `?` operator to check if the object exists before calling a method. This can make your code more streamlined if you don't need to do anything special when the object doesn't exist.
```

Comment:
```suggestion
Use the TypeScript `?` operator to check if the object exists before calling a method. This can make your code more streamlined if you don't need to do anything special when the object doesn't exist.
```